### ISSUE TITLE
Migrate from Mailchimp to Intercom

### ISSUE THREAD

**Issue by mattttt** [raintank/customer-experience#6]
We'd like to migrate from Mailchimp to Intercom.io for early access signups:

![earlyacces-intercom](https://cloud.githubusercontent.com/assets/2886187/8753185/ab34626e-2c87-11e5-91a4-3534be9ca14b.png)

We can use this angular wrapper for communicating with Intercom:
https://github.com/gdi2290/angular-intercom

**Comment by mattttt** [comment]
Ideally, the form submission on the website could resemble what we sent to mailchimp currently (included below), or something else entirely. The main important parts at the moment is being able to pass the $email and pieces of metadata added to the form ( ```value="new-home-callout-module"``` )

```
<form data-parsley-validate action="http://raintank.us9.list-manage.com/subscribe/post?u=332a8b5085fde6390a4408bb1&amp;id=e6455ee744" method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" class="inline-form newsletter-form">           
     <fieldset>
          <div style="position: absolute; left: -5000px;"><input type="text" name="b_332a8b5085fde6390a4408bb1_1d7e889d23" tabindex="-1" value=""> </div>
          <input type="email" placeholder="e-mail address" name="EMAIL" id="mce-EMAIL" class="callout-cta" required >
          <input type="hidden" name="SOURCE" value="new-home-callout-module">    <!-- value tells us which form was filled out -->
           <button type="submit" value="Subscribe" name="subscribe" id="mc-embedded-subscribe" class="btn primaryAltCTA">Get Early Access</button>
     </fieldset>
</form>
```

**Comment by mattttt** [comment]
@woodsaj: Chatted with @nopzor1200 on this Thursday night, and it 's currently a high priority. Before making the request here, I bounced it off @torkelo to see if it was a big/small request, and he seemed to think it was pretty straightforward.

In short, I'd much rather put something like this in place rather than build something strictly for the website.

**Comment by woodsaj** [comment]
I am not following what you want here.

The users are submitting a form from the www.raintank.io site for signup?  From your diagram i assume that you want the form posted to app.raintank,io backend?  But then you talk about using angular-intercom for sending the email.  You can't use angular from the backend.

I dont know anything about intercom.  What does it do. From your diagram it looks like i need to post it some data and it will asynchronously call some http callback service i also need to have running?

**Comment by mattttt** [comment]
> From your diagram i assume that you want the form posted to app.raintank,io backend?

Correct, currently we post to mailchimp, but we want to take them out of the mix.

> But then you talk about using angular-intercom for sending the email. You can't use angular from the backend.

Hrm, that makes sense. Ultimately, we just want to (1) make sure those email addresses get into Intercom and (2) avoid is having our API key publicly available in the JS.

Intercom will be the main customer communication channel, handling email and live chat communications initially. Long-term, it might also be a place where we can automatically trigger emails and notices based on event data, but that's something we'll evaluate later. 

Does that help give some better context?

https://doc.intercom.io/api/

**Comment by woodsaj** [comment]
What api method and what data should be posted to intercom?

Just conversations or do we want to create the new org (company) and user and then a coversation?

What data needs to provided to intercom? Do we need to add additional data to our own database?

**Comment by mattttt** [comment]
All we can post at this point is Email and the value field were sending to Mailchimp currently. We'll extend later when we know more (and they are actual clients with legs and names and hopes and dreams).

**Comment by woodsaj** [comment]
So this functionality doesnt  belong in the raintank backend at all.  The backend is the application and is designed to allow anyone to be able to deploy it.  

having a route to accept /earlyaccess requests is in no way a feature of the raintank/grafana  product and should live else where.

Sounds like we need to add a backend service for the www.raintank.io website.

**Comment by woodsaj** [comment]
https://github.com/raintank/website-v2/pull/16

Created a backend for the raintank.io website.

You can test it out, by running `nodejs app.js -p build/dev` instead of `grunt connect`
The site will then be available on localhost:3000

You will need to change the APPID and APIKEY in earlyaccess.js to test if it actually sends data to intercom.
